torch/_inductor/codegen/cpp.py

```diff
@@ -1049,45 +1224,187 @@ def to_dtype(x, dtype):
         return self
 
 
+class CppTile2DKernelChecker(CppVecKernelChecker):
+    """
+    We only addresss a narrow set of situations. All of the following conditions should be met:
+    1. There are only loads and stores.
```

Comment:
How will this extend to kernels that do compute?  Can we do this in a more general way?

Reply:
Generalize the implementation and added compute support too.